### PR TITLE
Add calendar to homepage

### PR DESCRIPTION
**Pull request by ianyong**
VirtualBoxVM_A4Ev18gk53

TODOs:
* Add a button to jump to today.
* Migrate the user task cards over from the `react-slick` library to the `swiper` library.
  * This should also fix the issue of the user task cards bugging out when changing date.
* Clean up user task card styling.
* Add toggle button to user task cards.

### CODE REVIEW

**Review by jialin7878 — APPROVED** [review]
looks amazing!! shld tapping on a date setSelectedDate to that day?

File: frontend/package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "date-fns": "^2.25.0",
     "immer": "^9.0.6",
     "js-cookie": "^3.0.1",
+    "moment": "^2.29.1",
```
Comment (sebastiantoh) [inline frontend/package.json:21]: nit: can we use `date-fns` instead? It's already added

Comment (ianyong) [reply frontend/package.json:21]: Hmm, does `date-fns` have an equivalent function as `moment`'s `calendar`?

Comment (ianyong) [reply frontend/package.json:21]: There is [this](https://date-fns.org/v2.25.0/docs/formatDistanceToNow) but it doesn't seem to be as customisable.

Comment (ianyong) [reply frontend/package.json:21]: Ok!

> Moment is no longer maintained and is much more bloated than dayjs (or any other date packages)

I wasn't aware of that.

**Pull request merged**